src/include/server/mir/input/key_mapper.h

```diff
@@ -30,14 +32,21 @@ namespace input
 {
 class Keymap;
 
+class KeyMapperObserver
+{
+public:
+    virtual ~KeyMapperObserver() = default;
+    virtual void leds_set(MirInputDeviceId id, KeyboardLeds leds) = 0;
+};
+
```

Comment:
This isn't a KeyMapper observer. It is a LED state observer.

We should be keeping the KeyMapper interface unchanged. The ObserverRegister can be part of another interface implemented by XkbMapper.

Also, wouldn't it be better to register with the device `id` and only send notifications to the corresponding device. (And not notify all devices most of which drop the notification.

Reply:
Yup I agree :+1: